### PR TITLE
feat(vats): facilitate launching additional PSMs

### LINKED ISSUE #6021: BLDer DAO can start PSMs for more reference stable interchain assets
## What is the Problem Being Solved?

After bootstrap, the BLDer DAO should be able to select more reference stable tokens and start PSM contracts for them.

## Description of the Design

core-eval ("big hammer") proposal to
  - add an interchain asset to the VBANK
  - start a PSM instance for that asset, with debt limit 0

## Security Considerations

Should the econ committee be able to do this?
_p.s. no_

## Test Plan

end-to-end test, ideally in a loadgen style automated harness

cc @dtribble @rowgraus @Tartuffo

### PR DESCRIPTION
**Pull request by dckc**
goal:
closes: #6021

## Description

This exposes the `startPSM` and `makeAnchorAsset` functions to the core-eval scope; as a result, a proposal to start a PSM for DAI is just a few lines.

TODO:
 - [x] handle naming issues around multiple PSMs

### Security Considerations

TODO:
 - [x] refine permit of example proposal
 - [x] set default fee limit to 0 IST

### Documentation Considerations

@rowgraus should we put something about this process in some Inter Protocol docs? Or just wait until it's time to discuss an actual proposal with the community?

### Testing Considerations

I did `local-chain` end-to-end integration testing, as noted [below](https://github.com/Agoric/agoric-sdk/pull/6142#issuecomment-1239630723)

### CODE REVIEW

**Comment by dckc** [comment]
## How to introduce additional PSMs to the psmCharter?

@dtribble @Chris-Hibbert bootstrap wires up "the" PSM to `psmCharter` for governing minting limits and such. When we start another PSM, I just realized we need to introduce it to the `psmCharter` somehow too.

File: packages/inter-protocol/test/psm/gov-add-psm-permit.json [hunk 1]
```diff
@@ -0,0 +1,52 @@
+{
+  "consume": {
+    "agoricNamesAdmin": true,
+    "bankManager": true,
+    "board": true,
+    "chainStorage": true,
+    "zoe": "zoe",
+    "feeMintAccess": "zoe",
+    "economicCommitteeCreatorFacet": "economicCommittee",
+    "chainTimerService": "timer"
```
Comment (dtribble) [inline packages/inter-protocol/test/psm/gov-add-psm-permit.json:12]: why does it need the timer service?  governance?

File: packages/inter-protocol/test/psm/gov-add-psm.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+/** @param {unknown} permittedPowers see gov-add-psm-permit.json */
+const main = async permittedPowers => {
+  console.log('starting PSM:', DAI);
+  const {
+    consume: { feeMintAccess: _, ...restC },
+    ...restP
+  } = permittedPowers;
+  const noMinting = { consume: restC, ...restP };
+  await Promise.all([
+    startPSM.makeAnchorAsset(noMinting, {
```
Comment (dtribble) [inline packages/inter-protocol/test/psm/gov-add-psm.js:37]: How can it make a PSM without minting power? all the PSMs will need that

File: packages/inter-protocol/test/psm/gov-add-psm-permit.json [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+      "psm": true,
+      "mintHolder": true
+    }
+  },
+  "instance": {
+    "consume": {
+      "economicCommittee": true
+    },
+    "produce": {
+      "$ISSUE": "These names assume there can be only one psm",
```
Comment (dtribble) [inline packages/inter-protocol/test/psm/gov-add-psm-permit.json]: Are these intended to remain?

**Review by dtribble — COMMENTED** [review]
I have some questions in the comments.

File: packages/inter-protocol/test/psm/gov-add-psm-permit.json (hunk 1 above)
Comment (dckc) [reply packages/inter-protocol/test/psm/gov-add-psm-permit.json:12]: yes.

in `startPSM.js`:

```js
  const governorTerms = await deeplyFulfilledObject(
    harden({
      timer: chainTimerService,
      economicCommittee,
...
  );
```

File: packages/inter-protocol/test/psm/gov-add-psm-permit.json (hunk 3 above)
Comment (dckc) [reply packages/inter-protocol/test/psm/gov-add-psm-permit.json:30]: no.

File: packages/inter-protocol/test/psm/gov-add-psm.js (hunk 2 above)
Comment (dckc) [reply packages/inter-protocol/test/psm/gov-add-psm.js:37]: This line is only making the anchor asset (issuer, ...). We don't need minting power for that.

File: packages/inter-protocol/test/psm/gov-add-psm-permit.json (hunk 3 above)
Comment (dckc) [reply packages/inter-protocol/test/psm/gov-add-psm-permit.json]: no. hence https://github.com/Agoric/agoric-sdk/pull/6142#issuecomment-1237629267 and #6146 and such.

File: packages/inter-protocol/test/psm/gov-add-psm.js (hunk 2 above)
Comment (dtribble) [reply packages/inter-protocol/test/psm/gov-add-psm.js:37]: Ah. I saw it invoking "starPSM" and thought it was starting it

**Comment by Chris-Hibbert** [comment]
I think #6146 addressed
 * [x]  handle naming issues around multiple PSMs
Can we check that off in the top-comment, or is there more to be done for naming?

**Comment by dckc** [comment]
connection to the `psmCharter` is working, finally:

```
2022-09-07T16:33:46.716Z block-manager: block 28 begin
2022-09-07T16:33:46.721Z SwingSet: vat: v1: issuer DAI: new Promise
2022-09-07T16:33:46.722Z SwingSet: vat: v1: brand DAI: new Promise
2022-09-07T16:33:46.727Z SwingSet: vat: v1: starting PSM: { keyword: 'DAI', decimalPlaces: 18, denom: 'ibc/toydai', proposedName: 'Maker DAI' }
XS snapshot written to /home/connolly/projects/agoric-sdk/packages/cosmic-swingset/t1/n0/data/ag-cosmos-chain-state/xs-snapshots/47d5177d084b205d875cf75dec79225aa57cf2e61d2584
0f4baa9f1aa4af9da5.gz : 1635995 bytes compressed, 7494503 raw
2022-09-07T16:33:47.902Z block-manager: block 28 commit
2022-09-07T16:33:51.736Z block-manager: block 29 begin
2022-09-07T16:33:51.897Z block-manager: block 29 commit
2022-09-07T16:33:56.749Z block-manager: block 30 begin
XS snapshot written to /home/connolly/projects/agoric-sdk/packages/cosmic-swingset/t1/n0/data/ag-cosmos-chain-state/xs-snapshots/0a106938f822316b12246b919fa522475b154b526938f5
1914863e2f0475b33f.gz : 1555397 bytes compressed, 7046479 raw
2022-09-07T16:33:58.057Z block-manager: block 30 commit
2022-09-07T16:34:01.759Z block-manager: block 31 begin
2022-09-07T16:34:02.874Z SwingSet: vat: v20: PSM Starting Object [Alleged: DAI brand] {} { denominator: { brand: Object [Alleged: IST brand] {}, value: 100n }, numerator: { br
and: Object [Alleged: DAI brand] {}, value: 100n } }
XS snapshot written to /home/connolly/projects/agoric-sdk/packages/cosmic-swingset/t1/n0/data/ag-cosmos-chain-state/xs-snapshots/75d15629896ebebc681d8bf282946327b7749da2578774
1ea2e72abdd9d22f10.gz : 1690238 bytes compressed, 7657983 raw
2022-09-07T16:34:03.158Z block-manager: block 31 commit
2022-09-07T16:34:06.772Z block-manager: block 32 begin
2022-09-07T16:34:06.868Z block-manager: block 32 commit
2022-09-07T16:34:11.783Z block-manager: block 33 begin
2022-09-07T16:34:11.922Z block-manager: block 33 commit
2022-09-07T16:34:16.798Z block-manager: block 34 begin
2022-09-07T16:34:16.970Z block-manager: block 34 commit
2022-09-07T16:34:21.815Z block-manager: block 35 begin
2022-09-07T16:34:21.927Z SwingSet: vat: v11: psmCharter: adding instance { minted: Object [Alleged: IST brand] {}, anchor: Object [Alleged: DAI brand] {} }
2022-09-07T16:34:21.937Z SwingSet: vat: v1: started PSM: { options: { anchorOptions: { keyword: 'DAI', decimalPlaces: 18, denom: 'ibc/toydai', proposedName: 'Maker DAI' } }, W
antStableFeeBP: 1n, GiveStableFeeBP: 3n, MINT_LIMIT: 0n }
2022-09-07T16:34:21.964Z block-manager: block 35 commit
```

**Comment by dckc** [comment]
@Chris-Hibbert and co, PTAL.

> I think #6146 addressed "handle naming issues around multiple PSMs"
> Can we check that off in the top-comment, or is there more to be done for naming?

There was some wiring up the `psmCharter`, but it's done now and checked off.

**Comment by dckc** [comment]
re docs: I can imagine the chainStorage and agoricNames stuff should be documented.

for now:


```
$ agoric follow :published.agoricNames.instance
[
  [
    "economicCommittee",
    slot(0,"Alleged: InstanceHandle"),
  ],
  [
    "psm-IST-AUSD",
    slot(1,"Alleged: InstanceHandle"),
  ],
  [
    "psm-IST-DAI",
    slot(2,"Alleged: InstanceHandle"),
  ],
]
```

```
$ agd query vstorage children published.psm.IST
children:
- AUSD
- DAI
```

File: packages/inter-protocol/src/psm/psmCharter.js [hunk 4]
```diff
@@ -46,7 +49,9 @@ export const start = async (zcf, privateArgs) => {
     );
   };
 
-  const makeOfferFilterInvitation = () => {
+  /** @param {Instance} instance */
```
Comment (dckc) [inline packages/inter-protocol/src/psm/psmCharter.js:52]: @turadg I'm interested to get some training on using `Instance` types parameterized by the contract `start` function and such.

**Comment by dckc** [comment]
oops! I force-pushed after requesting review. I hope I didn't make too much of a mess.

File: packages/vats/src/core/boot-psm.js [hunk 5]
```diff
@@ -5,10 +5,17 @@ import {
   makeAnchorAsset,
   startPSM,
   PSM_MANIFEST,
+  PSM_GOV_MANIFEST,
+  startPSMCharter,
 } from '@agoric/inter-protocol/src/proposals/startPSM.js';
+import * as startPSMmod from '@agoric/inter-protocol/src/proposals/startPSM.js';
+import * as ERTPmod from '@agoric/ertp';
```
Comment (Chris-Hibbert) [inline packages/vats/src/core/boot-psm.js:12]: Capitalization nit: shouldn't these be `startPSMMod` and `ERTPMod`?

**Review by Chris-Hibbert — APPROVED** [review]
LGTM

File: packages/vats/src/core/boot-psm.js (hunk 5 above)
Comment (dckc) [reply packages/vats/src/core/boot-psm.js:12]: could be. if this version doesn't pass CI, I'll tweak it.

File: packages/inter-protocol/src/psm/psmCharter.js (hunk 4 above)
Comment (turadg) [reply packages/inter-protocol/src/psm/psmCharter.js:52]: check out https://github.com/Agoric/agoric-sdk/compare/type-instance

**Pull request merged**